projects/components/src/stepper/stepper.component.ts

```diff
@@ -203,6 +211,16 @@ export class StepperComponent implements AfterContentInit {
   private isLastStep(stepper: MatStepper): boolean {
     return stepper.selectedIndex === stepper.steps.length - 1;
   }
+
+  /**
+   * Check if all the steps in between the selected index and the index are completed.
+   * Mandatory for linear stepper when navigating forward.
+   */
+  private areStepsInBetweenCompleted(selectedIndex: number, index: number): boolean {
```

Comment:
`selected index` here is the currently active index, right? I think we should rename it to `active index` and `index` can be renamed to `selected index`

Reply:
@23nobody Kept the naming in sync with what material is using I believe.

Reply:
I think we should move the `isGoingForward` code in this function itself.
and change the condition above
`this.isLinear  ? this.areStepsInBetweenCompleted(this.stepper.selectedIndex, index) : true`

Reply:
Updated.